c/public/lib/source/ttread/ttread.c

```diff
@@ -820,21 +993,551 @@ static void hmtxRead(ttrCtx h) {
+    int iTouch2;            /* second touched or reference point that has delta */
+    int iFirstDeltaPt;
+    int iContour;
+    int p;                  /* used as for loop index below */
+
+    for (iContour = 0; iContour < nContours; iContour++) {
+        iStartPt = ranges[iContour].begPt - ptBase;
+        iEndPt = ranges[iContour].endPt - ptBase;
+
+        iPt = iStartPt;
+        /* search first point which has delta */
+        while (iPt <= iEndPt && !hasDelta[iPt])
+            iPt++;
+
+        if (iPt <= iEndPt) {
+            iTouch1 = iPt;          /* first touched point which has delta. */
+            iFirstDeltaPt = iPt;    /* store the first point which has delta. */
+
+            iPt++;
+
+            /* search next touched point that has a delta */
+            while (iPt <= iEndPt) {
+                if (hasDelta[iPt]) {
+                    iTouch2 = iPt;
+
+                    /* interpolate intermediate points between iTouch1 and iTouch2.
+                     do it for x then for y coord. */
+                    gvarInterpolateIntermCoord(iTouch1 + 1, iTouch2 - 1, iTouch1, iTouch2, xorig, xDeltas);
+                    gvarInterpolateIntermCoord(iTouch1 + 1, iTouch2 - 1, iTouch1, iTouch2, yorig, yDeltas);
+
+                    iTouch1 = iPt;
+                }
+                iPt++;
+            }
+
+            /* if we have only single delta, shift all points with same difference as given by iTouch1. */
+            if (iFirstDeltaPt == iTouch1) {
+                Fixed deltaX = xDeltas[iTouch1];
+                Fixed deltaY = yDeltas[iTouch1];
+
+                for (p = iStartPt; p <= iEndPt; p++) {
+                    if (p != iTouch1) {
+                        xDeltas[p] = deltaX;
+                        yDeltas[p] = deltaY;
+                    }
+                }
+            } else {
+                /* interpolate remaining points at the end and beginning of the contour. */
+                gvarInterpolateIntermCoord(iTouch1 + 1, iEndPt, iTouch1, iFirstDeltaPt, xorig, xDeltas);
+                gvarInterpolateIntermCoord(iTouch1 + 1, iEndPt, iTouch1, iFirstDeltaPt, yorig, yDeltas);
+
+                if (iFirstDeltaPt > 0) {
+                    gvarInterpolateIntermCoord(iStartPt, iFirstDeltaPt - 1, iTouch1, iFirstDeltaPt, xorig, xDeltas);
+                    gvarInterpolateIntermCoord(iStartPt, iFirstDeltaPt - 1, iTouch1, iFirstDeltaPt, yorig, yDeltas);
+                }
+            }
+        }
+    }
+}
+
+/* apply glyph variation data to points in a glyph
+ * if nContours < 0, the glyph is a compound glyph
+ * delta values for a component apply to all its points */
+static void applyGlyphVariationDeltas(ttrCtx h, GID gid, int nContours, int nPoints, int nTotalPoints,
+                                      int ptBase, glyfCoord *coords, ptRange *ranges) {
+    Fixed imStart[VF_MAX_AXES];
+    Fixed imEnd[VF_MAX_AXES];
+    Fixed peakTupleCoords[VF_MAX_AXES];
```

Comment:
Please explicitly zero out `imStart`, `imEnd`, & `peakTupleCoords` -- Xcode's static analysis thinks it sees scenarios where at least `imStart` & `peakTupleCoords` might get used before they're initialized.

Reply:
added `memset` calls